buildenv/jenkins/common/build.groovy

```diff
@@ -529,10 +538,50 @@ def archive_diagnostics() {
             }
         }
         // Note: to preserve the files ACLs set _OS390_USTAR=Y env variable (see variable files)
-        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' | sed 's#^./##' | pax -wzf ${diagnosticsFilename}"
-    } else {
-        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' -o -name ddr_info -o -name j9ddr.dat -o -name superset.dat -o -name '*.annt.h' -o -name '*.stub.h' -o -name '*.dSYM' -o -path '*/make-support/failure-logs/*' | sed 's#^./##' | tar -zcvf ${diagnosticsFilename} -T -"
+        archiveCmd = "pax -wzf ${diagnosticsFilename}"
     }
+
+    def findDiagnosticsCmd = findCrashDataCmd
+    findDiagnosticsCmd += " -o -type d -name ddr_info -prune"
+    findDiagnosticsCmd += " -o -name j9ddr.dat"
+    findDiagnosticsCmd += " -o -name superset.dat"
+    findDiagnosticsCmd += " -o -name '*.annt.h'"
+    findDiagnosticsCmd += " -o -name '*.stub.h'"
+    findDiagnosticsCmd += " -o -path '*/make-support/failure-logs/*'"
```

Comment:
Are any of these useful only for debugging JVM crashes? I imagine it's the case for DDR stuff (including, I assume, `ddr_info` and `j9ddr.dat`), but can somebody confirm? And I'm not sure what the rest of these things are

Reply:
I'm not sure there's much value in capturing `j9ddr.dat`, `superset.dat` or the `.annt` or `.stub` files.
`ddr_info` is being pruned here so it won't be include (we need to be mindful of operator precedence) with that fragment.
OpenJDK makefiles capture command lines, etc. in `failure-logs` so those might be useful (they're relatively small files).

Reply:
With `-prune`, `find` will still find the `ddr_info` directory. It just won't descend into it. Compare:
```
$ find openj9-openjdk-jdk8/build/ -type d -name ddr_info -o -name j9jvmti.txt
openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/ddr_info
openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/ddr_info/targets/j9jvmti.txt
```
vs.
```
$ find openj9-openjdk-jdk8/build/ -type d -name ddr_info -prune -o -name j9jvmti.txt
openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/ddr_info
```
Descending into the directory is pointless since the directory itself is already listed to be included in the archive, and it will be included recursively. For example, take a look at [this crash](https://openj9-jenkins.osuosl.org/job/Build_JDK19_x86-64_mac_Nightly/127/consoleText), where the files being added to the archive beneath `ddr_info` do not match any of the other patterns that the `find` command was looking for:
```
[2022-12-01T01:08:36.763Z] build/macosx-x86_64-server-release/vm/ddr_info/
[2022-12-01T01:08:36.763Z] build/macosx-x86_64-server-release/vm/ddr_info/targets/
[2022-12-01T01:08:36.763Z] build/macosx-x86_64-server-release/vm/ddr_info/targets/ddrgentest.txt
[2022-12-01T01:08:36.763Z] build/macosx-x86_64-server-release/vm/ddr_info/targets/omrport_obj.txt
[2022-12-01T01:08:36.763Z] build/macosx-x86_64-server-release/vm/ddr_info/targets/j9jvmti.txt
...
```
So that's why I put `-prune` there, and also on the other directories that I want to include in their entirety.

However, I haven't checked yet to make sure that `pax -wzf ...` will also include directories recursively. If it doesn't, then I'll have to replace `-type d -path '...' -prune` with `-path '.../*'`.

Reply:
They're originally from https://github.com/eclipse-openj9/openj9/pull/9670

Reply:
Alright, so I guess out of these lines that leaves only `ddr_info` and `make-support/failure-logs`.

@keithc-ca, should we keep collecting `ddr_info`? Based on your earlier comment, I'm not totally clear on your position on that

Reply:
OK, well, I've removed the others and kept `ddr_info` for now. It's easy to remove it later if we want to